sdk/aqueduct/artifacts/numeric_artifact.py

```diff
@@ -126,6 +131,7 @@ def bound(
         equal: Optional[float] = None,
         notequal: Optional[float] = None,
         severity: CheckSeverity = CheckSeverity.WARNING,
+        lazy: bool = False,
```

Comment:
should we use `execution_mode` here?

Reply:
I also used to have `execution_mode ` here, but then I thought it may be easier for the users to type `lazy=True` instead of `execution_mode=ExecutionMode.LAZY`.

Reply:
I see..does `ExecutionMode` get exposed to user in other places? If so I feel it might be fine to keep consistent and force user to always use `ExecutionMode` when needed, otherwise I agree the current way is better so that the user don't need to access `ExecutionMode` at all

Reply:
Curious on how others think as well @kenxu95

Reply:
`ExecutionMode` isn't exposed to the user elsewhere, so I think `lazy=True` is probably easier.

Reply:
yeah agreed, this way user don't  need another import and code looks cleaner